src/Provider/DefaultSmsProvider.php

```diff
@@ -52,15 +55,54 @@ public function __construct(ConfigFactoryInterface $config_factory, ModuleHandle
   /**
    * {@inheritdoc}
    */
-  public function send(SmsMessageInterface $sms, array $options = array()) {
-    // Check if a preferred gateway is specified in the $options.
-    if (isset($options['gateway'])) {
-      $gateway = SmsGateway::load($options['gateway']);
+  public function queue(SmsMessageEntityInterface $sms_message) {
+    $gateway = $this->getGateway($sms_message);
+    if ($gateway->getSkipQueue()) {
+      $this->send($sms_message, []);
+      return;
+    }
+
+    if (!$sms_message->getGateway()) {
+      // @fixme getGateway being falsy is undocumented...
+      $sms_message->setGateway($this->getDefaultGateway());
+    }
+
+    if ($count = $sms_message->validate()->count()) {
+      throw new SmsException(sprintf('Can not queue SMS message because there are %s validation error(s).', $count));
+    }
+
+    // Split messages to overcome gateway limits.
+    $max = $gateway->getMaxRecipientsOutgoing();
+    $recipients_all = $sms_message->getRecipients();
+    if ($max > 0 && count($recipients_all) > $max) {
+      foreach ($sms_message->chunkByRecipients($max) as $sms_message) {
+        $sms_message->save();
+      }
     }
```

Comment:
The reason its here is so we can reduce memory usage. So when the message is initially queued we can say 'Hey your shit didnt get saved.'. Whereas doing large operations in the cron job could fail on this. The cron job focuses on sending operations, and optionally deletes the queued message if ret=0, so these are memory reductions only.

If we chunk with `save`, then when the cronjob comes around, it needs to delete the saved message and save new ones.

I got your point, but `send`  should be for the business of sending. Whereas `queue` is in the business of readying for #cronjob.

Perhaps its better to rename the `send` method to something that is not dev-facing. "dispatcher"?. Then recommend devs use `queue`? (or rename queue to send?). All without making the existing `send` method protected, just in case.